core/account-id/src/lib.rs

```diff
@@ -104,6 +104,24 @@ impl AccountId {
         !self.is_system() && !self.contains('.')
     }
 
+    /// Returns `true` if the account id can be viewed as an Ethereum address directly
+    ///
+    /// See [Ethereum account](https://ethereum.org/en/developers/docs/accounts/) for more details.
+    ///
+    /// ## Examples
+    /// ```
+    /// use near_account_id::AccountId;
+    ///
+    /// let eth_addr: AccountId = "0x06012c8cf97bead5deae237070f9587f8e7a266d".parse().unwrap();
+    /// assert!(eth_addr.is_ethereum_address());
+    ///
+    /// let non_eth_addr: AccountId = "0x".parse().unwrap();
+    /// assert!(!non_eth_addr.is_ethereum_address());
+    /// ```
+    pub fn is_ethereum_address(&self) -> bool {
+        self.starts_with("0x") && self.len() == 42
```

Comment:
Shouldn't this also check that the characters after the `0x` only come from the hex alphabet?